### PR TITLE
fix(echo): Trigger properly off Github Pull Requests

### PR DESCRIPTION
**Pull request by justinrlee**
If you send Github webhooks for PRs, they don't trigger anything.

Draft cause I haven't built tests yet.

### CODE REVIEW

File: echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java [hunk 1]
```diff
@@ -63,12 +63,15 @@ public void handle(Event event, Map postedEvent) {
...
+    // Default to .ref, then to pull_request.head.ref, then to "" (empty string)
+    if (githubWebhookEvent.ref == null && githubWebhookEvent.pull_request.head.ref == null) {
       event.content.put("branch", "");
+    } else if (githubWebhookEvent.ref == null) {
+      event.content.put("branch", githubWebhookEvent.pull_request.head.ref);
     } else {
       event.content.put("branch", githubWebhookEvent.ref.replace("refs/heads/", ""));
     }
-    event.content.put("repoProject", githubWebhookEvent.repository.owner.name);
+    event.content.put("repoProject", githubWebhookEvent.repository.owner.login);
```
Comment (staciegravy) [inline echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java]: How does this change affect things down the line?

**Comment by justinrlee** [comment]
This could probably be refactored to use `Optional` as part of https://github.com/spinnaker/echo/pull/486, but for now it should work.

**Comment by ethanfrogers** [comment]
Personally, I'd like to see this cleaned up to mimic #486. It will make it a lot easier to encapsulate the logic for each case if each item is in it's own method. Also, it will read _so much_ cleaner.

**Comment by justinrlee** [comment]
@ethanfrogers tada?

File: echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java [hunk 2]
```diff
@@ -59,34 +66,38 @@ public void handle(Event event, Map postedEvent) {
...
-        objectMapper.convertValue(postedEvent, GithubWebhookEvent.class);
+    GithubWebhookEvent webhookEvent = null;
+    String eventType = "";
 
-    event.content.put("hash", githubWebhookEvent.after);
-    if (githubWebhookEvent.ref == null) {
-      event.content.put("branch", "");
-    } else {
-      event.content.put("branch", githubWebhookEvent.ref.replace("refs/heads/", ""));
+    // TODO: Detect based on header rather than body key
```
Comment (ethanfrogers) [inline echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java]: You should actually be able to `TODO` this as part of this PR. `event.details.requestHeaders` should have a copy of all the headers. You can probably do something like this:
```
String eventType = event.details.requestHeaders.getOrDefault("X-Github-Event", "")
```

File: echo-webhooks/src/test/groovy/com/netflix/spinnaker/echo/controllers/WebhooksControllerSpec.groovy [hunk 3]
```diff
@@ -431,7 +476,7 @@ class WebhooksControllerSpec extends Specification {
       event = it[0]
     }
 
-    event.content.hash == "0000000000000000000000000000000000000000"
+    event.content.hash == null
```
Comment (ethanfrogers) [inline echo-webhooks/src/test/groovy/com/netflix/spinnaker/echo/controllers/WebhooksControllerSpec.groovy]: This should be set back to the original value.

**Comment by justinrlee** [comment]
Adding flavor from conversation with Ethan:
Gate doesn't currently pass in the `x-github-event` header, so for now we will detect event type by looking at JSON structure / fields.

**Pull request merged**